file_util.go

```diff
@@ -56,33 +60,46 @@ func (util *snowflakeFileUtil) compressFileWithGzip(fileName string, tmpDir stri
 	return gzipFileName, stat.Size(), nil
 }
 
-func (util *snowflakeFileUtil) getDigestAndSize(src **bytes.Buffer) (string, int64) {
-	chunkSize := 16 * 4 * 1024
+func (util *snowflakeFileUtil) getDigestAndSizeForStream(stream **bytes.Buffer) (string, int64, error) {
 	m := sha256.New()
-	r := getReaderFromBuffer(src)
+	r := getReaderFromBuffer(stream)
+
 	for {
-		chunk := make([]byte, chunkSize)
+		chunk := make([]byte, fileChunkSize)
```

Comment:
dow e need to always 'make' a new chunk each time?

Reply:
You are correct, we can pull this out of the loop.